src/pages/views/PageDetails.tsx

```diff
@@ -59,22 +74,99 @@ export const PageDetails: React.FC<PageDetailsProps> = ({ id, params }) => {
+                        variables: {
+                          file: fileAttribute.value
+                        }
+                      })
+                    )
+                  );
+
+                  const attributesWithAddedNewFiles: AttributeValueInput[] = uploadFilesResult.reduce(
+                    (attributesWithAddedFiles, uploadFileResult, index) => {
+                      const attribute = data.attributesWithNewFileValue[index];
+
+                      errors = [
+                        ...errors,
+                        ...uploadFileResult.data.fileUpload.uploadErrors
+                      ];
+                      return [
+                        ...attributesWithAddedFiles,
+                        {
+                          file:
+                            uploadFileResult.data.fileUpload.uploadedFile.url,
+                          id: attribute.id,
+                          values: []
+                        }
+                      ];
+                    },
+                    []
+                  );
+
                   const result = await pageUpdate({
                     variables: {
                       id,
-                      input: createPageInput(data)
+                      input: createPageInput(data, attributesWithAddedNewFiles)
                     }
                   });
 
-                  return result.data.pageUpdate.errors;
+                  errors = [...errors, ...result.data.pageUpdate.errors];
+
+                  if (errors.length === 0) {
+                    const deleteAttributeValuesResult = await Promise.all(
+                      pageDetails?.data?.page?.attributes.map(
+                        existingAttribute => {
+                          const fileValueUnused =
+                            existingAttribute.attribute.inputType ===
+                              AttributeInputTypeEnum.FILE &&
+                            existingAttribute.values.length > 0 &&
+                            data.attributes.find(
+                              dataAttribute =>
+                                dataAttribute.id ===
+                                existingAttribute.attribute.id
+                            ).value.length === 0;
+
+                          if (fileValueUnused) {
+                            return deleteAttributeValue({
+                              variables: {
+                                id: existingAttribute.values[0].id
+                              }
+                            });
+                          }
+                        }
+                      )
+                    );
+
+                    deleteAttributeValuesResult.forEach(deleteValueResult => {
+                      errors = [
+                        ...errors,
+                        ...deleteValueResult.data.attributeValueDelete.errors
+                      ];
+                    });
+                  }
+
+                  return errors;
```

Comment:
If stuff mentioned above would be moved to separate functions, you get something like `
```
[...deleteAttributeValuesResultErrors, 
...attributesWithAddedNewFilesErrors, 
...pageUpdateErrors]
```

Would probably be easier in the future to see what's going on, rather than having an array that was mutated couple of times